#26890: Tweak release image and add "models" image to enable models to be run, especially through tt-installer

knauth opened the pull request:
Per issue #26890

I am still testing this out and fixing CI issues, but I wanted to open this today to get feedback on the bulk of the functionality.

tt-rkim committed:
#26890: Tidy up release container trigger workflow to enable iterating on release image

tt-rkim committed:
Nooo

knauth committed:
Create second image target

This image target, tt-metalium-devel, contains all the needed deps for
users to run and test models as well as perform limited metalium
development

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

knauth committed:
Target ttnn and models images to build

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

knauth committed:
Rename image targets

"devel" is inappropriate in the context of existing containers, so let's
go with ttnn and models, to match the existing tt-metalium-models
syntax in tt-installer.

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

knauth committed:
Run on faster CPU image

Let's get those build times down.

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

knauth committed:
Add image matrix to test and release

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

ttmchiou commented on .github/workflows/publish-release-image.yaml:
```diff
@@ -41,12 +41,12 @@ jobs:
         run: |
           echo "WHEEL_FILENAME=$(ls -1 *.whl)" >> $GITHUB_ENV
-          REPO_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-metalium-${{ matrix.os }}-release-amd64
+          REPO_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-metalium-${{ matrix.os }}-${{ matrix.image }}-amd64
```
is there a reason why our publish-release-image.yaml wouldn't do a release image and instead do a ttnn/models image?

knauth replied:
My thought process was that the existing release container is best used as a ttnn runtime, so I changed it to "ttnn" to reflect the new "models" container. Perhaps we could use "release-models" and "release-ttnn"?

knauth committed:
Use ubuntu-latest to build

Workaround some outgoing traffic thing on our runners

knauth committed:
Give the create-docker-release workflow read/write permissions

knauth committed:
Give build-artifact write permission

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

rayraykay committed:
Directly take in wheel artifact name

knauth committed:
Apply suggestion from @tt-rkim

Co-authored-by: Raymond Kim <109366641+tt-rkim@users.noreply.github.com>

knauth committed:
Rename images

release and release-models preserve backwards compatibility, and we can
reevaluate names when we do a deeper dive into the available containers.

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

knauth committed:
Give build and publish jobs proper permissions

Signed-off-by: June Knauth (tt) <jknauth@tenstorrent.com>

tt-rkim reviewed (approved):
@ttmchiou 

I'll approve because @knauth has confirmed we won't change the current release image for now, and just build the new one.
